src/experimental-bots/filler/orderSubscriberFiltered.ts

```diff
@@ -194,6 +206,38 @@ class OrderSubscriberFiltered extends OrderSubscriber {
 			});
 		}
 	}
+
+	private updateUserMarkets(
+		userId: string,
+		buffer: Buffer,
+		currentMarkets: Set<number>
+	): void {
+		const previousMarkets = this.userMarkets[userId] || new Set<number>();
+
+		const newMarkets = new Set<number>();
+		currentMarkets.forEach((marketIndex) => {
+			if (!previousMarkets.has(marketIndex)) {
+				newMarkets.add(marketIndex);
+			}
+		});
+
+		const removedMarkets = new Set<number>();
+		previousMarkets.forEach((marketIndex) => {
+			if (!currentMarkets.has(marketIndex)) {
+				removedMarkets.add(marketIndex);
+			}
+		});
+
+		this.userMarkets[userId] = currentMarkets;
+
+		for (const marketIndex of newMarkets) {
+			this.sendUserAccountUpdateMessage(buffer, userId, 'update', marketIndex);
+		}
```

Comment:
why do we only send updates for new markets? even if the markets are all the same we want to send the update with the new buffer

Reply:
i think this should be currentMarkets

Reply:
i'd agree with this